registrations/serializers.py

```diff
@@ -360,7 +363,50 @@ class Meta:
 
 class SignUpGroupSerializer(serializers.ModelSerializer):
     view_name = "signupgroup-detail"
-    signups = SignUpSerializer(many=True)
+
+    def get_fields(self):
+        fields = super().get_fields()
+        fields["signups"] = SignUpSerializer(
+            many=True, required=False, partial=self.partial
+        )
+        return fields
+
+    def _update_signups(self, instance, validated_signups_data):
+        for signup_data in validated_signups_data:
+            if not signup_data.get("id"):
+                continue
+            if not self.partial:
+                signup_data["signup_group"] = instance
+            signup = SignUp.objects.get(pk=signup_data["id"])
+            signup_serializer = SignUpSerializer(
+                instance=signup,
+                data=signup_data,
+                context=self.context,
+                partial=self.partial,
+            )
+            signup_serializer.update(signup_serializer.instance, signup_data)
+
+    def update(self, instance, validated_data):
+        errors = {}
+
+        if (
+            "registration" in validated_data
+            and instance.registration != validated_data["registration"]
+        ):
+            errors["registration"] = _(
+                "You may not change the registration of an existing object."
+            )
+
+        if errors:
+            raise serializers.ValidationError(errors)
```

Comment:
You could have this kind of logic in `validate_registration` or `validate`. :thinking: